### PR TITLE
[WIP] Remove templates from user (and talk) pages when disenrolled

### PR DESCRIPTION
**Pull request by xiaobaioo7**
## What this PR does
When user is dis-enrolled from course, it makes updates to user page and user talk page to remove existing enrollment-in-course templates (if any). This intends to revert the addition to the pages when enrolling the user in course.

Issue #2842 

## Open questions and concerns
What are you thoughts about testing this?

### CODE REVIEW

**Comment by xiaobaioo7** [comment]
Hmm.. have been fixing the build, it looks like the build now has 2 blocking issues with class length over sized. How would you approach that?
image

**Comment by ragesoss** [comment]
That can be handled editing `.rubocop.yml` to either change the limit or exclude those files from the class length rule... or by refactoring some of the functionality of those large classes into new, if you see a way to do that cleanly.

**Comment by xiaobaioo7** [comment]
Thanks @ragesoss , I would think refactoring the large classes are the ultimate goal (and this is what rubocop would be used for). I would like to take another look into the effort. But for now, I temporarily excluded the 2 large classes from `.rubocop.yml`.

**Comment by xiaobaioo7** [comment]
About the 2 large classes, here are some of my thoughts on refactoring them:

- `UsersController`: we may extract the logic for user enrollment and dis-enrollment into a separate controller something like `UserEnrollmentsController`
- `WikiCourseEdits`: how about extracting the enrollment related logic into a separate class like `WikiCourseEnrollmentEdits`?

**Comment by ragesoss** [comment]
Thanks @xiaobaioo7! I think those sound like the right approaches to refactoring this.

**Comment by xiaobaioo7** [comment]
Thanks @ragesoss ! With an attempt to separate the enrollment logic from the 2 large classes into new ones, it seems to be passing the class max length rule from rubocop. In addition to the refactoring, I made some minor changes. Please see my inline comments.

File: lib/wiki_course_edits.rb [hunk 1]
```diff
@@ -204,7 +145,7 @@ def homewiki_assignments_grouped_by_article
   end
 
   def update_assignments_for_article(title:, assignments_for_same_article:)
-    return if WikiApi.new(@home_wiki).redirect?(title)
+    return if @wiki_api.redirect?(title)
```
Comment (xiaobaioo7) [inline lib/wiki_course_edits.rb]: Consider to use the variable from the constructor instead.

File: lib/wiki_course_edits.rb [hunk 2]
```diff
@@ -101,7 +85,6 @@ def remove_assignment(assignment:)
 
   private
 
-  # rubocop:disable Metrics/CyclomaticComplexity
```
Comment (xiaobaioo7) [inline lib/wiki_course_edits.rb]: Will this work still by removing the comment for rubocop at this line and one below to disable/enable Metrics/CyclomaticComplexity?
Running rubocop locally with the comments in place seems to produce offenses.

**Comment by ragesoss** [comment]
This looks like it was pretty close to being complete, but I must not have looked at it again after the refactor because it was still marked as [WIP]. This will probably be useful for someone else if they take up the issue.

**Pull request closed without merging**